### ISSUE TITLE
idevicebackup2: restoring from backups of earlier iOS versions?

### ISSUE THREAD

**Issue by jmpolom** [libimobiledevice/libimobiledevice#1202]
Should restoring a backup made from an earlier version of iOS than what's currently on the phone work? I attempted to restore a backup made from a device with iOS 14.6 onto a device with iOS 14.7.1 but it failed, citing a version issue. I updated the phone the backup was made from to iOS 14.7.1 and made a new backup. Restoring from the backup taken from iOS 14.7.1 (same version as the phone the backup was being restored onto) succeeded.

**Comment by nikias** [comment]
Yes this works, the backup system on the device has migrators to update older version specific data to the latest.
What's the version error you get? If it's about a "backup protocol version exchange" then it's not about the backup but an outdated libimobiledevice. So you need to build it from source.
